moved to #54

Dqu1J opened the pull request.

Dqu1J committed:
Rewriting structure renderer part 1

Dqu1J committed:
Formatting fixes

Dqu1J committed:
Update license

Dqu1J committed:
uh oh

Dqu1J committed:
face culling

Brian-Wuest commented on src/main/java/com/wuest/prefab/structures/render/PrefabBakedModel.java:
```diff
@@ -0,0 +1,30 @@
+    public void emitBlockQuads(BlockRenderView blockView, BlockState state, BlockPos pos, Supplier<Random> randomSupplier, RenderContext context) {
+        context.pushTransform((quad -> {
+            Direction dir = quad.cullFace();
+            if (dir == null) return true;
```
The body of this "if" should be wrapped in curly braces as well as a blank line above "if" and below the closing curly brace.

Brian-Wuest commented on src/main/java/com/wuest/prefab/structures/render/StructureRenderHandler.java:
```diff
@@ -64,61 +68,66 @@ public static void setStructure(Structure structure, Direction assumedNorth, Str
         }
+
+        if (structure != null && mc.world != null) {
+            renderSetup(mc.world, mc.player);
```
Static function calls should be pre-pended with the class name to make it clearer who owns the static function.

Brian-Wuest commented on src/main/java/com/wuest/prefab/structures/render/StructureRenderHandler.java:
```diff
@@ -134,75 +143,75 @@ public static void renderPlayerLook(PlayerEntity player, HitResult src, MatrixSt
-                    StructureRenderHandler.currentConfiguration.houseFacing);
-
-            return StructureRenderHandler.renderComponentInWorld(world, subBlock, entityVertexConsumer, matrixStack, subBlockPos);
+            renderBlocks(vertexBuffer, pos, camera.getPos(), stack);
```
Static function calls should be pre-pended with the class name to make it clearer who owns the static function.

Brian-Wuest reviewed (changes requested):
See comments for formatting items to address.

Dqu1J committed:
Fix occlusion culling

Dqu1J committed:
Formatting fixes part 2

Dqu1J committed:
fix ground z-fighting

Dqu1J committed:
render at separate blockRenderView, fixes face culling

Brian-Wuest commented on src/main/java/com/wuest/prefab/structures/render/StructureRenderHandler.java:
```diff
@@ -195,7 +196,7 @@ private static void renderBlockModel(BlockRenderView world, BakedModel bakedMode
     private static void renderBlocks(VertexBuffer vertexBuffer, BlockPos pos, Vec3d cameraPos, MatrixStack matrixStack) {
         matrixStack.push();
-        matrixStack.translate(pos.getX()-cameraPos.getX(), pos.getY()-cameraPos.getY(), pos.getZ()-cameraPos.getZ());
+        matrixStack.translate(pos.getX()-cameraPos.getX(), pos.getY()-cameraPos.getY()+0.001, pos.getZ()-cameraPos.getZ());
```
Needs extra spaces around the math operations in this method. They are difficult to notice otherwise.

If you are using IntelliJ and used the default keyboard layout you can press Alt + Ctrl + L to format the document.

Brian-Wuest reviewed (changes requested):
See formatting comments

Dqu1J committed:
Formatting fixes part 3

Dqu1J committed:
Don't render outside the world height

Dqu1J committed:
WBOIT part 1, not yet working

Brian-Wuest committed:
Dependency update for 1.18

Brian-Wuest committed:
Updated gradle

Brian-Wuest committed:
updating target to Java 17

Brian-Wuest committed:
Fixing compilation errors

Brian-Wuest committed:
Updating missing author

Brian-Wuest committed:
Fix issues with bed entities

Brian-Wuest committed:
Update mineshaft generation

We will now generate 21 levels above the world bottom. This is customizable by dimension.

Brian-Wuest committed:
Fixing mineshaft generation for moderate house

Brian-Wuest committed:
Fixing textures for blueprints without complex optons

Brian-Wuest committed:
Fixing scan of modern library

Brian-Wuest committed:
Fixing monster masher

Dqu1J commented on src/main/java/com/wuest/prefab/mixins/GameRendererMixin.java:
```diff
@@ -0,0 +1,17 @@
+public class GameRendererMixin {
+    @Inject(method = "loadShaders", at = @At("RETURN"))
+    private void loadShaders(ResourceManager manager, CallbackInfo ci) {
+        ShaderManager.loadShaders(manager);
```
Probably not going to use WBOIT, shader related code going to be removed.

Brian-Wuest committed:
Adding missing recipes

Brian-Wuest committed:
comment cleanup

Commit:
Revert "WBOIT part 1, not yet working"

Dqu1J committed:
Merge branch '1.17.1' into main

Brian-Wuest committed:
Version bump because I forgot to before.

Brian-Wuest committed:
Dependency version bump

Brian-Wuest committed:
Fixing netherite recipes

Brian-Wuest committed:
Mod version bump

Brian-Wuest committed:
Updating fabric api version

Brian-Wuest committed:
Adding new watch tower option

Brian-Wuest committed:
making watchtower more expensive to justify contents

Brian-Wuest committed:
Adding carpet to list of later blocks

Brian-Wuest committed:
Fixing issues with tile entities.

Brian-Wuest committed:
Adding restaurant and juice shop

Brian-Wuest committed:
Fixing issue with scanning entities.

Brian-Wuest committed:
Fixing scans for new buildings

Brian-Wuest committed:
Updating translations

Brian-Wuest committed:
Adding Quartz-Crete

This includes the main Block plus: Bricks, Pillar, Chiseled, Stairs, Slab, Smooth Block, Smooth Stairs, Smooth Slab, Wall (New!), Smooth Wall (Also New!) variants

Brian-Wuest committed:
Updating watch tower options

Also adding pictures for new structures

Brian-Wuest committed:
Updating modern buildings options for new buildings

Brian-Wuest committed:
Adding advanced and moderate modern building items

This is the first part to split out the modern buildings item.

Brian-Wuest committed:
Adding compressed and double compressed quartz-crete

Brian-Wuest committed:
Adding translations

For new moderate and advanced modern item as well as the compressed and double compressed quartz-crete

Brian-Wuest committed:
Adding modern moderate house

Brian-Wuest committed:
Updating dependencies

Brian-Wuest committed:
Adding image for moderate modern house

Brian-Wuest committed:
Breaking out modern building item

Brian-Wuest committed:
Switching modern moderate house to use quartz-crete

Brian-Wuest committed:
Removing configuration for loft house. it is always available now

Loft house was also updated to use quartz-crete instead of quartz

Dqu1J committed:
Merge branch 'main' of https://github.com/Brian-Wuest/MC-Prefab-Fabric into Brian-Wuest-main

Dqu1J committed:
Merge branch 'Brian-Wuest-main'

Dqu1J commented:
Moved to #54
